.github/workflows/linux-build.yml

```
@@ -32,12 +31,28 @@ jobs:
     steps:
       - name: Checkout Code
         uses: actions/checkout@v2
+      - name: Configure environment
+        run: |
+          TAG=$(git rev-parse --short HEAD)
+          echo "TAG=$TAG" >> $GITHUB_OUTPUT
+          echo "NAME=openocd64-$TAG" >> $GITHUB_ENV
       - name: Install required packages (apt-get)
         run: |
           sudo apt-get update
           sudo apt-get install libusb-1.0-0 libusb-1.0-0-dev
       - run: ./bootstrap
-      - run: ./configure --enable-remote-bitbang --enable-jtag_vpi --enable-ftdi-cjtag
+      - run: ./configure --enable-remote-bitbang --enable-jtag_vpi --enable-ftdi-cjtag --prefix /tmp/${{ env.NAME }}
       - run: make -j`nproc`
-      - run: file src/openocd | grep 64-bit
-      - run: src/openocd --version
+      - name: Check that we built something
+        run: |
+          file src/openocd | grep 64-bit
+          src/openocd --version
+      - name: Package
+        run: |
+          make install
+          tar zcvf ${{ env.NAME }}.tgz -C /tmp ${{ env.NAME }}
+      - name: Upload Artifact
+        uses: actions/upload-artifact@v3
+        with:
+          name: ${{ env.NAME }}
+          path: ${{ env.NAME }}.tgz
```

This results in a .zip package that contains a .tgz package that finally contains the built openocd.

For that reason, please consider removing the "Package" step.

```suggestion
      - name: Upload Artifact
        uses: actions/upload-artifact@v3
        with:
          name: ${{ env.NAME }}
          path: /tmp/${{ env.NAME }}
```

---

https://docs.github.com/en/actions/using-workflows/storing-workflow-data-as-artifacts says "billing is calculated based on the size of the uploaded artifact and not the size of the zip file." I don't actually knows how pays, presumably the RISC-V Foundation. But whoever it is, we save them money by compressing the data before uploading it. I agree it's annoying that this means you have a tgz that is zipped.

---

That page also says: "GitHub Actions usage is free for standard GitHub-hosted runners in public repositories", which I believe applies to this public riscv-openocd repository.

In any case, the artifact size reduction w.r.t. potential billing is a good argument (even though it probably does not make any difference in this repo at the moment). If you'd like to keep the tgz compression step in the workflow, please add there just a short comment to explain the reason.